components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +143,30 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+// See https://github.com/google/re2/wiki/Syntax for RE2 syntax
+bool IsValidConfig(const wchar_t* config) {
+  if (!config) {
+    return false;
+  }
+
+  std::string config_str;
+  if (!base::WideToUTF8(config, wcslen(config), &config_str)) {
+    VLOG(1) << "Unable to convert config to std::string. Is config valid?";
+    return false;
+  }
+
+  // Check for invalid characters
+  // low ascii (excluding CR/LF), #, high ascii
+  if (re2::RE2::PartialMatch(
+          config_str,
+          R"([\x00-\x09]|[\x0b-\x0c]|[\x0d-\x1f]|[\x23]|[\x7f-\xFF])")) {
```

Comment:
I wonder if this is the equivalent of `base::IsStringASCII`

Reply:
Basically this is making sure something is not in ASCII range 0 - 9, 11, 12, 14-31, 35 (`#`), or 126+

10 and 13 allowed because config has CRLF

Reply:
I suspect that's what `base::IsStringASCII` checks for
https://source.chromium.org/chromium/chromium/src/+/main:base/strings/string_util_unittest.cc;l=595;drc=90cac1911508d3d682a67c97aa62483eb712f69a

Reply:
Would need to see what the masks evaluate to:
https://source.chromium.org/chromium/chromium/src/+/main:base/strings/string_util_impl_helpers.h;l=164-167;drc=90cac1911508d3d682a67c97aa62483eb712f69a

I think what we have is OK - but can look further if it's a blocker

Reply:
I think it's better to use a well tested existing utility than a regex if it has what we're looking for